### PR TITLE
feat(rentry): init

### PR DESCRIPTION
**Pull request by thismoon**
## 🎉 Theme for Rentry 🎉

Rentry.co is a markdown paste service with preview, custom urls and editing.[¹](https://rentry.co/what)

## 💬 Additional Comments 💬

there are classes but they were never used, instead the website is themed with hardcoded color values from [bootstrap.min.css](https://rentry.org/static/css/bootstrap.min.css?v=85). so i had to theme every element individually

this specific table header in ["how"](https://rentry.co/how) is colored differently
image
image
because it's different than normal tables in user created pages
image
image

users can set text with custom colors using this syntax:
```py
 %red% Colored Text %%
```
and while in theory it's possible to color them
```css
    .color-change[style="color:red"]{
      color: @red !important
    }
    .color-change[style="color:green"]{
      color: @green !important
    }
    .color-change[style="color:blue"]{
      color: @blue !important
    }
```
there are [too many colors](https://rentry.org/rentry-text-colors). it's possible to color the main ones,
note: users still can still set custom colors with hex codes
```c
 %#ACBDEF% Colored Text Hex %%
```
theming those colors would open a can of worms because this
```py
 %violet% !~green; Combine With Text Color ~! %%
```
results in this
```html
<span style="text-decoration-line:underline;text-decoration-color:green;color:violet">Combine With Text Color</span>
```
the only exception i did is coloring `grey` and `#039205` because they're used in an [official page](https://rentry.co/claim-guide)

i added a regex for pages ending in `/raw` , pages starting with `https://rentry.co/static/` and pages on the subdomain `export3.rentry.co` so they default to `prefers-color-scheme` (device theme) instead of `@lightFlavor` (default light theme because the way dark theme is set is by adding a class to the root but without it it's light theme, and these pages don't have a toggle button for dark theme but they follow the device theme instead)
also i think it's possible to only keep `@base` and `@text` only in the catppuccin mixin in that section but i'm not sure if i should do it

~~the accent color was never used­­~~ i forgot that `::selection{background-color: fade(@accent-color, 30%);}` exists

the syntax highlighting is using CodeMirror. and while [there is a theme for it](https://github.com/catppuccin/codemirror), it's not present in [tips-and-tricks.md](https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#how-do-i-theme-code-blocks--syntax-highlighting). also i think its [outdated](https://github.com/catppuccin/codemirror/issues/2). so i themed everything manually (according to the default light theme css)

## 🗒 Checklist 🗒

- [x] I have read and followed Catppuccin's [submission guidelines](https://github.com/catppuccin/userstyles/blob/main/docs/userstyle-creation.md).
- [x] I have made a new directory underneath `/styles/<name-of-website>` containing the contents of the [`/template`](https://github.com/catppuccin/userstyles/blob/main/template/) directory.
  - [x] I have ensured that the new directory is in **lower-kebab-case**.
  - [x] I have followed the template and kept the preprocessor as [LESS](https://lesscss.org/#overview).
- [x] I have made sure to update the
      [`userstyles.yml`](https://github.com/catppuccin/userstyles/blob/main/scripts/userstyles.yml)
      file with information about the new userstyle.
- [x] I have included the following files:
  - [x] `catppuccin.user.css` - all the CSS for the userstyle, based on the
        template.
  - [x] `preview.webp` - composite image of all four individual flavor screenshots (taken with the default accent color of mauve) stitched together, generated via [Catwalk](https://github.com/catppuccin/catwalk).

### CODE REVIEW

**Comment by thismoon** [comment]
i didn't set the title as `Feat/rentry`. was that autogenerated? anyway i fixed it.

**Comment by uncenter** [comment]
> i didn't set the title as `Feat/rentry`. was that autogenerated? anyway i fixed it.

GitHub autofills it from the name of your branch.

File: styles/rentry/catppuccin.user.css [hunk 1]
```diff
@@ -0,0 +1,526 @@
...
+    .highlight .vc,
+    .highlight .vg,
+    .highlight .vi,
+    .highlight .vm {
+      color: @lavender;
+    }
+    .highlight .si,
+    .highlight .sr {
+      color: @red;
+    }
```
Comment (uncenter) [inline styles/rentry/catppuccin.user.css:448]: Nesting please.

Comment (uncenter) [reply styles/rentry/catppuccin.user.css:448]: Also please make sure you haven't just taken their colors and made them Catppuccin- syntax highlighting needs to match how our style guide is, so look at how code looks in (paste it in to) VS Code or Neovim or whatever your editor is and try to match the classes as well as you can.

File: styles/rentry/catppuccin.user.css [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+      color: @red;
+      background-color: fade(@red, 25%);
+    }
+    .admonition.danger > .admonition-title:before,
+    .admonition.error > .admonition-title:before {
+      @svg: escape(
+        '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="@{red}" viewBox="0 0 16 16"><path d="M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM8 4a.905.905 0 0 0-.9.995l.35 3.507a.552.552 0 0 0 1.1 0l.35-3.507A.905.905 0 0 0 8 4zm.002 6a1 1 0 1 0 0 2 1 1 0 0 0 0-2z"/></svg>'
+      );
+      content: url("data:image/svg+xml,@{svg}");
+    }
```
Comment (uncenter) [inline styles/rentry/catppuccin.user.css:306]: Some nesting would be nice here.

File: styles/rentry/catppuccin.user.css [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+    .cm-comment {
+      color: @pink !important;
+    }
+    .cm-hr {
+      color: @surface2;
+    }
+    // markdown
+    .cm-header {
+      color: @text;
+    }
```
Comment (uncenter) [inline styles/rentry/catppuccin.user.css:224]: Same thing with my other syntax highlighting comment, make sure it actually resembles our syntax highlighting ports.

**Review by uncenter — CHANGES_REQUESTED** [review]
In general, a lot of missed nesting and not very organized. I recommend grouping things into semantic sections based on where/what pages elements appear on, and utilize blank lines between selectors to make it easier to read and signal a shift in the content.

**Review by uncenter — CHANGES_REQUESTED** [review]
Delete/danger buttons (`.btn-danger`) are incorrectly themed by the default button styling, you'll want to add a section for `.btn-danger` below the `.btn` part.

File: styles/rentry/catppuccin.user.css (hunk 2 above)
Comment (thismoon) [reply styles/rentry/catppuccin.user.css]: this one should be solved with [da5f71b](https://github.com/catppuccin/userstyles/pull/1198/commits/da5f71bf52333406bef35ae7cf3780e2a9ad324a)

**Comment by thismoon** [comment]
how do i fix `Unoptimized SVG detected (catppuccin/optimized-svgs)`? do i put svgs into svgo? was this mentioned somewhere in the docs? i only found it in [this js file](https://github.com/catppuccin/userstyles/blob/main/scripts/lint/stylelint-custom/optimized-svgs.js) (which does use svgo)

**Comment by somerand0mcat** [comment]
> how do i fix `Unoptimized SVG detected (catppuccin/optimized-svgs)`?

your supposed to add ``fill="@{blue}"`` to ``path`` and not ``svg`` [like in the docs](https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md#how-do-i-theme-images-and-svgs)

**Comment by thismoon** [comment]
> @thismoon your supposed to add `fill="@{blue}"` to `path` and not `svg` [like in the docs](https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md?rgh-link-date=2024-08-15T18%3A14%3A00Z#how-do-i-theme-images-and-svgs)

thanks for the suggestion @somerand0mcat, but it didn't fix the issue
image

**Comment by GenShibe** [comment]
> > @thismoon your supposed to add `fill="@{blue}"` to `path` and not `svg` [like in the docs](https://github.com/catppuccin/userstyles/blob/main/docs/tips-and-tricks.md?rgh-link-date=2024-08-15T18%3A27%3A10Z#how-do-i-theme-images-and-svgs)
> 
> thanks for the suggestion @somerand0mcat, but it didn't fix the issue image

run `deno task lint:fix` and it'll optimize the svgs for you

**Comment by thismoon** [comment]
> run `deno task lint:fix` and it'll optimize the svgs for you

thanks! this fixed it. i forgot about [this part](https://github.com/catppuccin/userstyles/blob/main/docs/CONTRIBUTING.md#development-environment) in the docs

**Comment by uncenter** [comment]
> > run `deno task lint:fix` and it'll optimize the svgs for you
> 
> thanks! this fixed it. i forgot about [this part](https://github.com/catppuccin/userstyles/blob/main/docs/CONTRIBUTING.md?rgh-link-date=2024-08-15T19%3A42%3A42Z#development-environment) in the docs

Glad you figured it out. It can be confusing, are you in the Catppuccin Discord? We have a channel there exclusively for userstyles and userstyles help. Happy to answer any questions you ever have there!

**Comment by thismoon** [comment]
> Glad you figured it out. It can be confusing, are you in the Catppuccin Discord? We have a channel there exclusively for userstyles and userstyles help. Happy to answer any questions you ever have there!

yeah im in the discord server, i didn't know there was a forum thread for userstyles. will ask questions there next time. thanks

File: styles/rentry/catppuccin.user.css (hunk 1 above)
Comment (thismoon) [reply styles/rentry/catppuccin.user.css]: i tried to match the colors in [`1e1ec94` (#1198)](https://github.com/catppuccin/userstyles/pull/1198/commits/1e1ec9444e82f53af5578d20bf1fd4fb1e3fd99f) with the vscode(/ium) theme and it became more accurate however there are things i can't solve such as special characters (`(){}.<>=*:;`) being in the same color due to using the same class (`.p`) ([style-guide.md](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md#code-editors) says that symbols and atoms(?) should be red so i made it red)

if you notices any issues with the syntax highlighting please point them out

File: styles/rentry/catppuccin.user.css (hunk 3 above)
Comment (thismoon) [reply styles/rentry/catppuccin.user.css]: the [style guide](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md) is focused on code based syntax highlighting (terminals, code editors) but these are colors for (afaik) rentry specific syntax (i don't know other languages that support something specific like  ==marks== or how i am supposed to theme a [TOC] element). i guess the closest thing would be markdown but this seems like it's own thing

image
image
image

File: styles/rentry/catppuccin.user.css [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+@license MIT
+
+@preprocessor less
+@var select lightFlavor "Light Flavor" ["latte:Latte*", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha"]
+@var select darkFlavor "Dark Flavor" ["latte:Latte", "frappe:Frappé", "macchiato:Macchiato", "mocha:Mocha*"]
+@var select accentColor "Accent" ["rosewater:Rosewater", "flamingo:Flamingo", "pink:Pink", "mauve:Mauve*", "red:Red", "maroon:Maroon", "peach:Peach", "yellow:Yellow", "green:Green", "teal:Teal", "blue:Blue", "sapphire:Sapphire", "sky:Sky", "lavender:Lavender", "subtext0:Gray"]
+==/UserStyle== */
+
+@-moz-document domain('rentry.org'), domain('rentry.co') {
+  :root.dark-mode {
```
Comment (isabelroses) [inline styles/rentry/catppuccin.user.css:19]:
```suggestion
  .dark-mode {
```

**Review by isabelroses — DISMISSED** [review]

**Review by somerand0mcat — APPROVED** [review]
nice work! next time you should run ``deno task lint:fix && deno task format`` to both fix any lint errors and format the code

File: scripts/userstyles.yml [hunk 5]
```diff
@@ -770,6 +770,13 @@ userstyles:
       app-link: "https://reddit.com"
     current-maintainers: []
     past-maintainers: [*jayylmao, *rubyowo]
+  rentry:
+    name: Rentry
```
Comment (uncenter) [inline scripts/userstyles.yml:774]:
```suggestion
  rentry.co:
    name: Rentry.co
```
This is how they brand themselves from what I can tell. Make sure to update the UserCSS metadata and directory name.

**Review by uncenter — CHANGES_REQUESTED** [review]

File: styles/rentry/catppuccin.user.css
Comment (uncenter) [inline styles/rentry/catppuccin.user.css:1]: Looking much much better. Do you mind adding comments for the types of tokens? E.g. constants, variables, keywords, strings, punctuation, operators, comments, etc.

Comment (uncenter) [reply styles/rentry/catppuccin.user.css:1]: Also, on those few remaining innaccuracies - I would go side by side with the playground on https://shiki.style/ (scroll down a bit, change the language to something like JavaScript and the theme to Catppuccin <insert flavor>) and compare the Shiki highlighting to how it looks on Rentry (paste the demo code Shiki has into a Rentry markdown code block).

**Comment by thismoon** [comment]
> [thismoon](https://github.com/thismoon) dismissed [isabelroses](https://github.com/isabelroses)’s [stale review](https://github.com/catppuccin/userstyles/pull/1198#pullrequestreview-2251802987)

what? i didn't do that

**Comment by thismoon** [comment]
I think i messed it up by renaming the branch
i will open a new pr i guess?

**Pull request closed without merging**